File: packages/core/src/__tests__/fixtures/sample.ts
/**
 * sample.ts — TypeScript fixture file for extractor tests.
 * This file contains various exported function patterns that the TS extractor must handle.
 */

// 1. Standard exported function declaration
export function add(a: number, b: number): number {
  return a + b
}

// 2. Exported async function
export async function fetchUser(id: string): Promise<{ name: string; email: string }> {
  // In a real implementation this would call an API
  return { name: 'Alice', email: 'alice@example.com' }
}

// 3. Exported arrow function (const)
export const multiply = (a: number, b: number): number => a * b

// 4. Exported arrow function (async)
export const delay = async (ms: number): Promise<void> => {
  await new Promise((resolve) => setTimeout(resolve, ms))
}

// 5. Exported function with optional param and default
export function greet(name: string, greeting = 'Hello'): string {
  return `${greeting}, ${name}!`
}

// 6. Exported function that throws
export function divide(a: number, b: number): number {
  if (b === 0) throw new Error('Division by zero')
  return a / b
}

// 7. Exported class
export class Calculator {
  private history: number[] = []

  add(a: number, b: number): number {
    const result = a + b
    this.history.push(result)
    return result
  }

  getHistory(): number[] {
    return [...this.history]
  }
}

// NOT exported — should NOT be extracted
function internalHelper(x: number): number {
  return x * 2
}

// NOT exported — should NOT be extracted
const privateArrow = (s: string) => s.toLowerCase()

// Suppress unused variable warnings
void internalHelper
void privateArrow

File: packages/core/src/__tests__/fixtures/sample.py
"""
sample.py — Python fixture file for extractor tests.
Contains various function patterns that the Python extractor must handle.
"""

# 1. Simple function with type hints
def add(a: int, b: int) -> int:
    return a + b


# 2. Async function
async def fetch_user(user_id: str) -> dict:
    return {"name": "Alice", "email": "alice@example.com"}


# 3. Function with default parameter
def greet(name: str, greeting: str = "Hello") -> str:
    return f"{greeting}, {name}!"


# 4. Function with multiple return types (Union)
def divide(a: float, b: float) -> float:
    if b == 0:
        raise ValueError("Division by zero")
    return a / b


# 5. Function with no type hints
def legacy_add(a, b):
    return a + b


# 6. Function with *args
def sum_all(*args: int) -> int:
    return sum(args)


# 7. Class with methods
class Calculator:
    def __init__(self) -> None:
        self.history: list[int] = []

    def add(self, a: int, b: int) -> int:
        result = a + b
        self.history.append(result)
        return result

    def get_history(self) -> list[int]:
        return list(self.history)


# 8. Private-by-convention function (still extracted — Python has no true private)
def _internal_helper(x: int) -> int:
    return x * 2
